src/model.jl

```diff
@@ -1,3 +1,20 @@
+""" Abstract type for an abstract model formulated in the state space form
+"""
+abstract type AbstractStateSpaceModel <: AbstractMCMC.AbstractModel end
```

Comment:
Should this be in AdvancedPS?

Reply:
I think this should inherit from https://github.com/SciML/DifferenceEquations.jl/blob/14a50c6b6abacbd22e93122f7934b0f787ca721f/src/DifferenceEquations.jl#L11. I'm currently adding in all the nonlinear/arbitrary state space methods so this should be sufficient.

Reply:
Though we may need a wrapper that satisfies `<: AbstractMCMC.AbstractModel` anyway to meet the AbstractMCMC requirements

Reply:
It's a bit unfortunate that SciMLBase isn't as lightweight as one might assume and DifferenceEquations also depends on Optim (which itself depends on ForwardDiff). The large number of dependencies was one of the reasons why we replaced NonlinearSolve (which also depends on SciMLBase and ForwardDiff) with Roots again in Bijectors: https://github.com/TuringLang/Bijectors.jl/pull/202

Reply:
The Optim dependency shouldn't be there, I think, that's my bad. I'll remove that.

Reply:
Let me move it, but do we want to add SciMLBase as a dependency ?

Reply:
Maybe avoid depending on `SciMLBase` unless there is a very strong reason. In general, we would like to keep the Turing sampler packages lightweight.

Reply:
Okay, then let's stick with an internal interface and we'll do the interconnect later.